### ISSUE TITLE
RT 4.2.1148 mucks up MakerNotes for PENTAX K-1 files

### ISSUE THREAD

**Issue by TheQuo** [Beep6581/RawTherapee#3403]
I recently upgraded from PENTAX K10D to K-1, but the MakerNotes written by RT to the JPGs are somewhat mucked up.

**All files mentioned within this issue report are available from my dropbox:**
**[https://www.dropbox.com/sh/0yhyupwl32o04s5/AAAC1H2dk3OUXoal1Hvv4VL6a?dl=0](https://www.dropbox.com/sh/0yhyupwl32o04s5/AAAC1H2dk3OUXoal1Hvv4VL6a?dl=0)**

**Download and Version Information:**   (file: `Download and Version Information.txt`)
http://rawtherapee.com/releases_head/windows/RawTherapee_WinVista_64_Gtk3_Release_4.2.1148.zip

```
Branch: gtk3
Version: 4.2.1148
Changeset: 2bfb0acdd7ea202bb716e84585802d1875e41da4
Compiler: gcc 5.3.0
Processor: generic x86
System: Windows
Bit depth: 64 bits
Gtkmm: V3.18.0
Build type: release
Build flags: -m64 -mwin32 -mthreads -std=gnu++11 -mtune=generic -Werror=unused-label -fopenmp -Werror=unknown-pragmas -msse -msse2 -mwindows -fopenmp -Wno-aggressive-loop-optimizations -DNDEBUG -O3
Link flags: -m64 -mwin32 -mthreads -static-libgcc -mtune=generic -mwindows -s -O3
OpenMP support: ON
MMAP support: ON
```

**System Information:** See file `Systeminformation.png`
**General setting for Metadata processing:** See file `RawTherapee 4.2.1148.Metadata.Settings.png`
**General steps to reproduce this issue:**
I have two files in RAW-Format: `IMGP3191.DNG` and `IMGP2835.PEF`
For each file, I did the following:
1. Open these files in **Digital Camera Utility 5.5.1**, a PENTAX-supplied image-processing-software based on SILKYPIX.
2. Save file in JPG-Format: See files `IMGP3191.DNG.DigitalCameraUtility5.5.1.jpg` and `IMGP2835.PEF.DigitalCameraUtility5.5.1.jpg`
3. Dump the Metadata with [ExifTool](http://www.sno.phy.queensu.ca/~phil/exiftool/) and [exiv2](http://www.exiv2.org/).
See files:
- `IMGP3191.DNG.ExifTool.Dump.html`
- `IMGP3191.DNG.exiv2.Dump.txt`
- `IMGP2835.PEF.ExifTool.Dump.html`
- `IMGP2835.PEF.exiv2.Dump.txt`
- `IMGP3191.DNG.DigitalCameraUtility5.5.1.jpg.ExifTool.Dump.html`
- `IMGP3191.DNG.DigitalCameraUtility5.5.1.jpg.exiv2.Dump.txt`
- `IMGP2835.PEF.DigitalCameraUtility5.5.1.jpg.ExifTool.Dump.html`
- `IMGP2835.PEF.DigitalCameraUtility5.5.1.jpg.exiv2.Dump.txt`

The MakerNotes-Tags from these JPGs are recognized the same way as they were in K10D-files written by RT

1. Open `IMGP3191.DNG` in RT 4.2.1148, see file `IMGP3191.DNG.RawTherapee 4.2.1148.Metadata.png`
As you will see in the Metadata tab, EXIF and Makernotes folder are on the same level
2. Save file in JPG-format and dump Metadata. See files:

- `IMGP3191.DNG.RawTherapee.4.2.1148.jpg`
- `IMGP3191.DNG.RawTherapee.4.2.1148.jpg.ExifTool.Dump.html`
- `IMGP3191.DNG.RawTherapee.4.2.1148.jpg.exiv2.Dump.txt`

3. Open `IMGP2835.PEF` in RT 4.2.1148, see file `IMGP2835.PEF.RawTherapee 4.2.1148.Metadata.png`
As you will see in the Metadata tab,  the Makernotes are a subfolder of EXIF.
4. Save file in JPG-format and dump Metadata. See files:

- `IMGP2835.PEF.DigitalCameraUtility5.5.1.jpg`  
- `IMGP2835.PEF.DigitalCameraUtility5.5.1.jpg.ExifTool.Dump.html`

Here is the following error reported:
`**Warning: Bad MakerNotes offset for CameraSettingsUnknown**`
see file `IMGP2835.PEF.RawTherapee 4.2.1148.ExifTool.Error.png`
`IMGP2835.PEF.DigitalCameraUtility5.5.1.jpg.exiv2.Dump.txt` 

Here are the following errors reported:
```
   **Error: Offset of directory PentaxDng, entry 0x0205 is out of bounds: Offset = 0x00002b26; truncating the entry
   Error: Offset of directory PentaxDng, entry 0x0206 is out of bounds: Offset = 0x00002b4e; truncating the entry
   Error: Offset of directory PentaxDng, entry 0x0207 is out of bounds: Offset = 0x00002b7e; truncating the entry
   Error: Offset of directory PentaxDng, entry 0x0208 is out of bounds: Offset = 0x00002bfe; truncating the entry
   Error: Offset of directory PentaxDng, entry 0x0215 is out of bounds: Offset = 0x00002c22; truncating the entry
   Error: Offset of directory PentaxDng, entry 0x0216 is out of bounds: Offset = 0x00002c36; truncating the entry
   Error: Upper boundary of data for directory PentaxDng, entry 0x03ff is out of bounds: Offset = 0x00002820, size = 768, exceeds buffer size by 364 Bytes; truncating the entry**
```

I came across this, because I like to show the PENTAX-LensRec-Tags, which are, according to [http://www.sno.phy.queensu.ca/~phil/exiftool/TagNames/Pentax.html](http://www.sno.phy.queensu.ca/~phil/exiftool/TagNames/Pentax.html), located at 0x003F, with my favourite Imageviewer [IrfanView](http://www.irfanview.com/). In fullscreen-mode you specifiy the decimal equivalent (63 as $E63) in the options to display the LensType. I contacted the author and he is going to update his routines with the JPGs written by PENTAXs Digital Camera Utility as a reference.

Cheers Axel

**Comment by Hombre57** [comment]
@TheQuo 

I don't see the error message (I'm using the latest exiftool) but the Makernotes are not there anyway. So it's a +1 for solving this bug ASAP. I'm not assigning myself as I don't master this kind of stuff (jpeg, metadata) but will look into this sometime soon because I'm directly concerned ;)

**Comment by Beep6581** [comment]
@TheQuo @Hombre57  Is this still reproducible in dev?

**Comment by heckflosse** [comment]
@Beep6581 Didn't try it last time. But as I didn't solve it and @Hombre57 also didn't solve it I guess it is valid.

**Comment by TheQuo** [comment]
@Beep6581 Just checked with 5.0-r1-gtk3, behaviour remains the same.

**Comment by TheQuo** [comment]
Just checked with the newly released 5.4 (a big THANKS to everyone who made it possible!), but the MakerNotes are still missing in the resulting jpg, rawdata ist in dng-format.

**Comment by Beep6581** [comment]
Confirmed.
I would close it as depending on #3801

**Comment by Beep6581** [comment]
Closing as duplicate of #3801 - Adopt libexiv2 or ExifTool for metadata handling.

See also #4683
